### PR TITLE
Issue 2820 - Fix CI tests

### PR DESCRIPTION
**Pull request by aadhikar**
Bug Description: tickets/ticket47973_test.py test had failures in CI nightly runs

Fix Description: Fixed the failure and also changed the code to use DSLdapObject
moved the code into the schema test suite.

Relates: https://github.com/389ds/389-ds-base/issues/2820

Reviewed by: ??

### CODE REVIEW

File: dirsrvtests/tests/suites/schema/schema_reload_test.py [hunk 1]
```diff
@@ -8,10 +8,13 @@
...
 import os
+import re
+from lib389.tasks import *
 from lib389.topologies import topology_st as topo
-from lib389._constants import TASK_WAIT
+from lib389._constants import DEFAULT_SUFFIX
+from lib389._mapped_object import DSLdapObjects
+from lib389.idm.user import UserAccounts
 
 pytestmark = pytest.mark.tier1
```
Comment (aadhikar) [inline dirsrvtests/tests/suites/schema/schema_reload_test.py:19]: originally tickets/ticket47973_test.py was under tier2.

File: dirsrvtests/tests/suites/schema/schema_reload_test.py [hunk 2]
```diff
@@ -25,6 +28,143 @@
...
+    if moz:
+        log.error('case 2: MOZILLAATTRIBUTE is in the objectclasses list -- FAILURE')
+        assert False
+    else:
+        log.info('case 2: MOZILLAATTRIBUTE is not in the objectclasses list -- PASS')
+
+    Mozattr2 = "mozillaattribute"
+    log.info(f'Case 2-1: Use the custom schema with {Mozattr2}')
+    name = "test_user"
+    topo.standalone.add_s(Entry((f"cn={name},{DEFAULT_SUFFIX}", {
```
Comment (aadhikar) [inline dirsrvtests/tests/suites/schema/schema_reload_test.py]: Not able to use any ldapobject here, I tried using directly DSLdapObjects.create() but getting assertion error. Can do it using UnsafeExtensibleObjects but not sure about this.

Comment (droideck) [reply dirsrvtests/tests/suites/schema/schema_reload_test.py]: The issue is that you use 'Entry' object that will be deprecated one day - #3656 
I thing it's good enough if you'll just use direct add_s pyhton-ldap call:
https://www.python-ldap.org/en/python-ldap-3.3.0/reference/ldap.html#ldap.LDAPObject.add_s

**Review by droideck — CHANGES_REQUESTED** [review]
The rest looks good to me. But maybe @Firstyear will something to say...

File: dirsrvtests/tests/suites/schema/schema_reload_test.py (hunk 2 above)
Comment (aadhikar) [reply dirsrvtests/tests/suites/schema/schema_reload_test.py:158]: @droideck Hey I can change that, no issue but I am using Entry in checkTask() and it only accept that, any suggestion there?

File: dirsrvtests/tests/suites/schema/schema_reload_test.py [hunk 3]
```diff
@@ -25,6 +28,143 @@
...
+    topo.standalone.tasks.schemaReload(schemadir=topo.standalone.schemadir,
+                                              args={TASK_WAIT: False})
+
+    # While we wait for the task to complete search for our user
+    search_count = 0
+    while search_count < 10:
+    # Now check the user is still being returned
+    # Check if task is complete
+        assert user.exists()
+        task_entry = Entry(topo.standalone.tasks.dn)
```
Comment (aadhikar) [inline dirsrvtests/tests/suites/schema/schema_reload_test.py]: Will you use checkTask(topo.standalone.tasks.entry)[0] instead of this.

Comment (droideck) [reply dirsrvtests/tests/suites/schema/schema_reload_test.py]: `topo.standalone.tasks` refers to `Tasks()` object which is legacy.

You can use `Schema(DSLdapObject)` which has method `reload()` - it'll do the job.

```
    task = schema.reload(schema_dir=topo.standalone.schemadir)
    task.wait()
    assert task.get_exit_code() == 0
```

File: dirsrvtests/tests/suites/schema/schema_reload_test.py (hunk 2 above)
Comment (aadhikar) [reply dirsrvtests/tests/suites/schema/schema_reload_test.py]: @droideck Hi I have made the changes let me know if you think that is correct or not. Thanks!

**Comment by droideck** [comment]
Please, avoid 'git push **--force**' option as it's hard to follow what changes were maid.
Just add new commits as it's mentioned above:

`Add more commits by pushing to the fix-ticket47973 branch on aadhikar/389-ds-base.`

File: dirsrvtests/tests/suites/schema/schema_reload_test.py [hunk 4]
```diff
@@ -25,6 +30,145 @@
...
+    log.info('Test the searches still work as expected during schema reload tasks')
+
+    # Add a user
+    users = UserAccounts(topo.standalone, DEFAULT_SUFFIX)
+    user = users.create_test_user(uid=1)
+
+    # Run a schema_reload tasks while searching for our user.Since
+    # this is a race condition, run it several times.
+
+    topo.standalone.tasks.schemaReload(schemadir=topo.standalone.schemadir,
```
Comment (droideck) [inline dirsrvtests/tests/suites/schema/schema_reload_test.py]: As I mentioned before, we better replace legacy objects sooner than later.
Please, use `Schema.reload` option that I've described two comments before (you can check in your email notifications because it was removed from here as you've rebased)

**Comment by aadhikar** [comment]
@droideck opps, sure i will keep that in mind. Normal push was giving me some error i guess but sure I will completely avoid it.

**Review by droideck — COMMENTED** [review]
> @droideck opps, sure i will keep that in mind. Normal push was giving me some error i guess but sure I will completely avoid it.

No problem:) Ping me on IRC if you have some annoying issue and we'll try to figure it out.

File: dirsrvtests/tests/suites/schema/schema_reload_test.py [hunk 5]
```diff
@@ -56,18 +54,16 @@ def test_schema_reload_with_searches(topo):
...
 
     # While we wait for the task to complete search for our user
     search_count = 0
     while search_count < 10:
     # Now check the user is still being returned
     # Check if task is complete
         assert user.exists()
-        task_entry = topo.standalone.tasks.entry
-        if topo.standalone.tasks.checkTask(task_entry)[0]:
+        if task.get_exit_code() == 0:
```
Comment (aadhikar) [inline dirsrvtests/tests/suites/schema/schema_reload_test.py:66]: didn't use task.wait() here as the test needs it to be a race condition. Also made changes in other tests.

**Review by droideck — APPROVED** [review]
LGTM!
